src/dshould/basic.d

```diff
@@ -0,0 +1,328 @@
+    2.should.not.be(5);
+    5.should.equal(5);
+    5.should.not.equal(6);
+    5.should.not.be.equal.greater(6);
+    5.should.be.equal.greater(5);
+    5.should.be.equal.greater(4);
+    5.should.not.be.equal.greater(6);
+    5.should.be.smaller(6);
+}
+
+unittest
+{
+    (new Object).should.not.be(new Object);
+    (new Object).should.not.equal(new Object);
+    (new Object).should.not.be(null);
+    (cast(Object) null).should.be(null);
+
+    auto obj = new Object;
+
+    obj.should.equal(obj);
+    obj.should.be(obj);
+
+    class SameyClass
+    {
+        override bool opEquals(Object o) { return true; }
+    }
+
+    (new SameyClass).should.not.be(new SameyClass);
+    (new SameyClass).should.equal(new SameyClass);
+
+    (cast(void delegate()) null).should.be(null);
+}
+
+auto not(Should)(Should should) pure
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!([], "not");
+
+    return should.addWord!"not";
+}
+
+auto be(Should)(Should should) pure
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!(["not"], "be");
+
+    return should.addWord!"be";
+}
+
+void be(Should, T)(Should should, T expected, string file = __FILE__, size_t line = __LINE__)
+if (isInstanceOf!(ShouldType, Should) && !should.hasWord!"approximately")
+{
+    import std.format : format;
+
+    with (should)
+    {
+        allowOnlyWordsBefore!(["not"], "be");
+
+        enum isNullType = is(T == typeof(null));
+        // only types that can have toString need to disambiguate
+        enum isReferenceType = is(T == class) || is(T == interface);
+
+        auto got = should.got();
+
+        static if (hasWord!"not")
+        {
+            const refInfo = isReferenceType ? "different reference than " : "not ";
+
+            static if (isNullType)
+            {
+                check(got !is null, ": expected non-null", ", but got null", file, line);
+            }
+            else
+            {
+                check(
+                    got !is expected,
+                    format(": expected %s%s", refInfo, expected.quote),
+                    format(", but got %s", got.quote),
+                    file, line
+                );
+            }
+        }
+        else
+        {
+            const refInfo = isReferenceType ? "same reference as " : "";
+
+            static if (is(T == typeof(null)))
+            {
+                check(got is null, ": expected null", format(", but got %s", got.quote), file, line);
+            }
+            else
+            {
+
+                check(
+                    got is expected,
+                    format(": expected %s%s", refInfo, expected.quote),
+                    format(", but got %s", got.quote),
+                    file, line
+                );
+            }
+        }
+    }
+}
+
+auto equal(Should)(Should should)
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!(["not", "be", "greater", "smaller"], "equal");
```

Comment:
disallow `.greater.smaller` and `.smaller.greater`?

Reply:
`<>` is still D.